src/query/atomWithQuery.ts

```diff
@@ -43,10 +43,19 @@ export function atomWithQuery<
         typeof createQuery === 'function' ? createQuery(get) : createQuery
       let settlePromise: ((data: TData | null, err?: TError) => void) | null =
         null
-      const getInitialData = () =>
-        typeof options.initialData === 'function'
-          ? (options.initialData as InitialDataFunction<TQueryData>)()
-          : options.initialData
+
+      const getInitialData = () => {
+        let data: TQueryData | TData | undefined =
+          queryClient.getQueryData<TData>(options.queryKey)
+
+        if (!data && options.initialData)
+          data =
+            typeof options.initialData === 'function'
+              ? (options.initialData as InitialDataFunction<TQueryData>)()
+              : options.initialData
```

Comment:
```suggestion
        if (!data && options.initialData) {
          data =
            typeof options.initialData === 'function'
              ? (options.initialData as InitialDataFunction<TQueryData>)()
              : options.initialData
        }
```
sorry for the tiny styling thing. this is mostly preference but to make style consistent.
(I think there's an eslint rule. maybe we can try to enable it.)

Reply:
yep, fixed in https://github.com/pmndrs/jotai/pull/653/commits/d2599e20e2eea00b1b23f100b725da5b463236ce